Enyim.Caching/Memcached/MemcachedNode.cs

```diff
@@ -270,7 +273,7 @@ private class InternalPoolImpl : IDisposable
             /// <summary>
             /// A list of already connected but free to use sockets
             /// </summary>
-            private ConcurrentStack<PooledSocket> _freeItems;
+            private LinkedList<PooledSocket> _freeItems;

```

Comment:
Why not use ConcurrentDictionary? It will be more efficient for concurrent reads and writes, without having to manage the locking and mutexing all by ourselves?

Reply:
We need ordering, which a dictionary won't provide. Therefore, we would have to create a thread-safe pointer, which is more difficult to create and maintain than this solution

Reply:
But then, we can use list and concurrentdictionary and a combination to achieve this.